### PR TITLE
[Doc] Make runtimeEnvYAML more prominent in docs

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
When using Ray Jobs, it's better to specify the `runtime_env` in the Job spec, rather than in `ray.init()` in the entrypoint script. The reason is that it's simpler; in the first case the entrypoint script runs in the runtime env (as well as any tasks/actors), whereas in the second case only the tasks and actors run in the runtime env.

Some users are still using `runtime_env` in `ray.init()`. One possible reason is that it's not prominent enough in the docs.  This PR adds a reference to KubeRay RayJob in the Ray Core runtime env docs, and also adds a code snippet to make it more friendly for users to copy and paste.  

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by kevin85421** [comment]
The CI test `docs/readthedocs.com:anyscale-ray` failed.

**Review by kevin85421 — APPROVED** [review]

**Comment by architkulkarni** [comment]
Screenshot 2024-04-01 at 8 40 20 AM
Screenshot 2024-04-01 at 8 40 36 AM
Rendering looks OK, merging.

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md [hunk 1]
```diff
@@ -30,7 +30,20 @@ To understand the following content better, you should understand the difference
   * `rayClusterSpec` - Defines the **RayCluster** custom resource to run the Ray job on.
 * Ray job configuration
   * `entrypoint` - The submitter runs `ray job submit --address ... --submission-id ... -- $entrypoint` to submit a Ray job to the RayCluster.
-  * `runtimeEnvYAML` (Optional): A runtime environment that describes the dependencies the Ray job needs to run, including files, packages, environment variables, and more. Provide the configuration as a multi-line YAML string. See {ref}`Runtime Environments <runtime-environments>` for more details. _(New in KubeRay version 1.0.0)_
+  * `runtimeEnvYAML` (Optional): A runtime environment that describes the dependencies the Ray job needs to run, including files, packages, environment variables, and more. Provide the configuration as a multi-line YAML string. Example:
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md:33]:
```suggestion
  * `runtimeEnvYAML` (Optional): A runtime environment that describes the dependencies the Ray job needs to run, including files, packages, environment variables, and more. Provide the configuration as a multi-line YAML string.
  Example:
```

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md [hunk 2]
```diff
@@ -30,7 +30,20 @@ To understand the following content better, you should understand the difference
...
+
+    ```yaml
+    # [...]
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md:36]: Not sure if this adds any value. Is there some pseudo comment that would make this more useful? If not and this is common practice, then please disregard my comment.

**Review by angelinalg — APPROVED** [review]
Nice job!

File: doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md (hunk 2 above)
Comment (architkulkarni) [reply doc/source/cluster/kubernetes/getting-started/rayjob-quick-start.md]: Not sure either. I don't feel strongly about it so I removed it

**Pull request merged**